### PR TITLE
Fixed history command

### PR DESCRIPTION
**Pull request by syokkendesuyo**
- Display changed from UUID to player name
- More parts can be translated
 
image

### CODE REVIEW

**Comment by pop4959** [comment]
What is the history command?

File: src/lang/lwc_en.properties [hunk 1]
```diff
@@ -619,3 +619,13 @@ lwc.time.seconds=seconds
 lwc.time.early=less than a second
 lwc.time.unknown=Not yet known
 lwc.time.past=%time% ago
+
+#Supports single-byte characters only
```
Comment (pop4959) [inline src/lang/lwc_en.properties:623]: What do you mean by this comment? Is this an issue?

File: src/lang/lwc_jaJP.properties [hunk 2]
```diff
@@ -9,7 +9,7 @@
 public=共有
 private=個人
 password=パスワード
-kick_trap=KICKトラップ
+kick_trap=キックトラップ
```
Comment (pop4959) [inline src/lang/lwc_jaJP.properties]: Can you put miscellaneous translation fixes in a separate PR, as it doesn't directly pertain to this one.

File: src/main/java/com/griefcraft/modules/history/HistoryModule.java [hunk 3]
```diff
@@ -173,15 +175,21 @@ private void historyTool(LWC lwc, Player player, int x, int y, int z) {
...
     public void sendHistoryList(CommandSender sender, List<History> relatedHistory, int page, int maxPages, int historyCount) {
-        String format = "%4s%12s%12s%12s";
+        LWC lwc = LWC.getInstance();
+        String format = "%4s%16s%12s%10s";
 
         // Header
-        LWC.getInstance().sendLocale(sender, "lwc.history.list", "header", String.format(format, "Id", "Player", "Type", "Status"), "size", relatedHistory.size(),
-                "page", page, "totalpages", maxPages, "totalhistory", historyCount);
+        lwc.sendLocale(sender, "lwc.history.list",
+                "header", String.format(format, lwc.getLocaleMessage(sender, "lwc.history.id")[0], lwc.getLocaleMessage(sender, "lwc.history.player")[0], lwc.getLocaleMessage(sender, "lwc.history.type")[0], lwc.getLocaleMessage(sender, "lwc.history.status")[0]),
```
Comment (pop4959) [inline src/main/java/com/griefcraft/modules/history/HistoryModule.java]: Is there a way to clean this up to make the code more readable? The repeated `lwc.getLocaleMessage(sender, translation)[0]` looks very ugly here.

File: src/main/java/com/griefcraft/modules/history/HistoryModule.java [hunk 4]
```diff
@@ -173,15 +175,21 @@ private void historyTool(LWC lwc, Player player, int x, int y, int z) {
...
+                "size", relatedHistory.size(),
+                "page", page,
+                "totalpages", maxPages,
+                "totalhistory", historyCount
+        );
 
         // Send all that is found to them
         for (History history : relatedHistory) {
-            sender.sendMessage(String.format(format, ("" + history.getId()), history.getPlayer(), history.getType(), history.getStatus()));
+            sender.sendMessage(String.format(format, history.getId(), UUIDRegistry.getName(UUID.fromString(history.getPlayer())), lwc.getPlugin().getMessageParser().parseMessage(history.getType().toString().toLowerCase()), lwc.getPlugin().getMessageParser().parseMessage(history.getStatus().toString().toLowerCase())));
```
Comment (pop4959) [inline src/main/java/com/griefcraft/modules/history/HistoryModule.java]: Same here, a lot of repeated code. You could, for example, only get the message parser once, and re-use it.

**Review by pop4959 — CHANGES_REQUESTED** [review]
Overall good, just needs to be cleaned up a bit before merging this.

Still not sure what history is used for though. Maybe you can inform me on how you use it normally?

File: src/lang/lwc_en.properties (hunk 1 above)
Comment (syokkendesuyo) [reply src/lang/lwc_en.properties:623]: Format collapses.
This problem is caused by `Java.lang.String`.

Single byte:
image

Multi byte:
image

Comment (pop4959) [reply src/lang/lwc_en.properties:623]: Ah ok, so it's a String.format issue. Got it.

**Comment by pop4959** [comment]
Looks like the translations are still not completely reverted. Can you please put these in a new PR?

**Comment by syokkendesuyo** [comment]
The translation file has been rolled back.

File: src/lang/lwc_jaJP.properties [hunk 5]
```diff
@@ -619,4 +619,4 @@ lwc.time.second=秒
 lwc.time.seconds=秒
 lwc.time.early=現在
 lwc.time.unknown=フメイ
-lwc.time.past=%time% 経過
+lwc.time.past=%time% 経過
```
Comment (pop4959) [inline src/lang/lwc_jaJP.properties:622]: There still appears to be a diff here. Please see "Files changed" after you fix this to verify that there are no further changes so I can merge this.

**Review by pop4959 — CHANGES_REQUESTED** [review]

**Comment by pop4959** [comment]
Closing this since it's been over a month. Please feel free to open/re-open this PR if you update it with a clean diff.
